tasks/travis-pr-title-lint.js

```diff
@@ -72,25 +72,69 @@ module.exports = function(grunt) {
 
   function getPullRequestTitle() {
     return new Promise(function(resolve, reject) {
-      var pullRequestURLPath = '/repos/' +
-        process.env.TRAVIS_REPO_SLUG + '/pulls/' +
-        process.env.TRAVIS_PULL_REQUEST + '.json';
+      var pullRequestURLPath = '/' +
+        process.env.TRAVIS_REPO_SLUG + '/pull/' +
+        process.env.TRAVIS_PULL_REQUEST;
 
       grunt.log.writeln(
-        'Retrieving the pull request title from https://api.github.com' +
+        'Retrieving the pull request title from https://github.com' +
         pullRequestURLPath
       );
 
-      https.get({
-        host: 'api.github.com',
+      var req = https.get({
+        host: 'github.com',
         path: pullRequestURLPath,
         headers: {
           'User-Agent': 'mozilla web-ext grunt tasks',
         },
       }, function(response) {
+        if (response.statusCode < 200 || response.statusCode > 299) {
+          reject(new Error('Unexpected statusCode: ' + response.statusCode));
+          return;
+        }
+
         var body = '';
         response.on('data', function(data) {
-          body += data;
+          try {
+            body += data;
+
+            // Once we get the closing title tag, we can read
+            // the pull request title and
+            if (body.includes('</title>')) {
+              response.removeAllListeners('data');
+              response.emit('end');
+
+              var titleStart = body.indexOf('<title>');
+              var titleEnd = body.indexOf('</title>');
+
+              // NOTE: page slice is going to be something like:
+              // "<title> PR title by author · Pull Request #NUM · mozilla/web-ext · GitHub"
+              var pageTitleParts = body.slice(titleStart, titleEnd)
+                  .replace('<title>', '')
+                  .split(' · ');
+
+              // Check that we have really got the title of a real pull request.
+              var expectedPart1 = 'Pull Request #' +
+                process.env.TRAVIS_PULL_REQUEST;
+
+              if (pageTitleParts[1] === expectedPart1) {
+                // NOTE: the extracted title still contains the
+                // "by author" part, but it should not affect the
```

Comment:
@kumar303 actually this will affect the "changelog linting" if the resulting string  "pull request title + by author" exceeds the 72 character limit :-(